release_guidelines.md

```diff
@@ -8,73 +8,97 @@ We follow [semver](http://semver.org/) for versioning. Given a version number MA
 
 ## Creating a Release Candidate
 
-An example `<VERSION_RC>` could be `2.0.0-rc.1`
+An example release candidate version `<VERSION_RC>` could be `2.0.0-rc.1`
 
-* Create a release branch: `release/<version>`. Use the final version rather than a release candidate in the branch name.
-* Update the SDK package version in `package.json` to `<VERSION_RC>`
-* Update the change log following [this](http://keepachangelog.com/)
-* Increment `BASE_32_VERSION` in `webpack.config.babel.js` e.g. `AA` => `AB`
-  * You only need to do this once per release, *not* per release candidate
-* Build the dist files with `npm run build`
-* Run `aws s3 sync ./dist s3://onfido-assets-production/web-sdk/<BASE_32_VERSION>/ --exclude "*.html" --exclude "*.map" --acl public-read --delete`
-* [Deploying the release to S3 production](#deploying-the-release-to-S3-production)
-  * Run `aws s3 sync ./dist s3://onfido-assets-production/web-sdk-releases/<VERSION_RC> --exclude "*.html" --exclude "*.map" --acl public-read --delete`
-* Commit the above using the version as the commit message
-* Create a pull request from the release branch into master
-* [Update JSFiddle demo](#update-jsfiddle-demo)
-* On the release branch run `npm publish --tag next`
-  * Get the credentials from One Login
-  * Read about npm and release candidates [here](https://medium.com/@mbostock/prereleases-and-npm-e778fc5e2420)
-* Check the `latest` tag had not moved with `npm dist-tag ls onfido-sdk-ui`
-* Check you can install the package with `npm install onfido-sdk-ui@<VERSION_RC>`
-* Create a git tag with rc version on github, i.e `2.0.0-rc.1`:
-  * Pointing to the release commit on the release branch
-  * The release tag and title should be the version
-  * The description should be the change log entry for the release
-* Test the SDK deployment on surge associated with the PR
+1. Make sure all features for the forthcoming release have been merged to `development` branch.
+2. Checkout and pull `development` branch
+    - `git checkout development && git pull`
+3. Make sure README.md file has been updated
+4. Make sure CHANGELOG.md has been updated
+    - with new version number
+    - with all Public, Internal and UI changes
+5. Make sure MIGRATION.md has been updated, if applicable.
+6. Update the SDK package version in `package.json` to `<VERSION_RC>`
+7. Increment `BASE_32_VERSION` in `webpack.config.babel.js` e.g. `AA` => `AB`
+    - do it only if a breaking change is introduced between SDK and cross device client. This must be done only ONCE per release, *not* per release candidate
+8. Install npm dependencies
+    - `npm install`
+9. Build the dist files for release candidate
+    - `npm run build`
+10. Run `aws s3 sync ./dist s3://onfido-assets-production/web-sdk/<BASE_32_VERSION>/ --exclude "*.html" --exclude "*.map" --acl public-read --delete`
```

Comment:
what does this step do?

Reply:
I don't think we need to run this step anymore. Before we were using the links to this files to have the JSFiddle and code splitting running but now public path for prod points at `s3://onfido-assets-production/web-sdk-releases/<VERSION_RC>`

Reply:
Although this step would allow us to back up the different versions of the cross device separately.

Reply:
I now described this as `Deploy dist files to the incremented `<BASE_32_VERSION>` on AWS production`. Do you think it's descriptive enough?

Reply:
I updated it but please review if it's a really required step

Reply:
I just double checked with @rfreitas and he reckons we should keep this step as it will be easier to keep track of the different versions of the cross device that gets lazy loaded. We only need to push this if the base32 changes. Maybe the step should be:
`10. If <BASE_32_VERSION> has changed, run `aws s3 sync ./dist s3://onfido-assets-production/web-sdk/<BASE_32_VERSION>/ --exclude "*.html" --exclude "*.map" --acl public-read --delete`

Reply:
I don't understand. How is it different to step 11?

Reply:
@rfreitas can you extend on this please?

Reply:
the base32 is used only for the hosting of the cross device client. we only update it when something changes that can affect that flow.

step 11 is the hosting of every sdk version. Which at the time is consumed, transparently to the developer, by the lazy loading functionality. The lazy loading is not the cross device client. The cross device master actually uses lazy loaded modules. Therefore they need to updated for different reasons. But since these lazy loaded modules don't have the same constraints as the base32 one, we can upload it every time for every release (and we should, since it's simpler and less error prone).

if this is too complicated, we can just bump the base32 one for every release and we simplify the whole process (then we can actually merge the two), but we use up our base32 versions more quickly.

Reply:
Thanks, that makes sense now. I don't think we need to worry about using up the base32 versions so I would be happy to merge the steps.

@dawiddylowicz could you at least add a brief explanation of what steps 10 and 11 do?

Reply:
OK so as far as I understand, 'merging' is basically removing step 10. from `Publishing a release` section and step 3. from `Publishing a release` section, right?

Reply:
For now I added descriptions to these steps so that we know which one does. I think we can think about merging both steps into one as part of another ticket